main/main.ino

```diff
@@ -1037,106 +1049,105 @@ void loop() {
 
 #if defined(ESP8266) || defined(ESP32)
   if (WiFi.status() == WL_CONNECTED) {
+    ArduinoOTA.handle();
 #else
   if ((Ethernet.hardwareStatus() != EthernetW5100 && Ethernet.linkStatus() == LinkON) || (Ethernet.hardwareStatus() == EthernetW5100)) { //we are able to detect disconnection only on w5200 and w5500
 #endif
     failure_number_ntwk = 0;
-    if (client.connected()) {
+    if (!ProcessLock) { // If no blocking process we can handle MQTT processing
+      if (client.connected()) {
 #ifdef ZmqttDiscovery
-      if (!connectedOnce) pubMqttDiscovery(); // at first connection we publish the discovery payloads
+        if (!connectedOnce) pubMqttDiscovery(); // at first connection we publish the discovery payloads
 #endif
 
-      connectedOnce = true;
-      failure_number_ntwk = 0;
+        connectedOnce = true;
+        failure_number_ntwk = 0;
 
-      client.loop();
-
-#if defined(ESP8266) || defined(ESP32)
-      ArduinoOTA.handle();
-#endif
+        client.loop();
```

Comment:
Does this block, has this a callback, how long does it loop?

Reply:
No idea, what are you thinking about?